app/helpers/notifications_helper.rb

```diff
@@ -38,6 +38,18 @@ def filters
     }
   end
 
+  def repo_selected?
+    filters[:repo].present?
+  end
+
+  def repo_subscribed?
+    begin
+      return true if current_user.github_client.subscription("#{filters[:repo]}")
```

Comment:
I think this will kill the performance, as it's making an extra http request to the github api for every notification, we'll need to find another way to check this.

Reply:
It's actually not for every notification. It is only for a repository. So, the request is made only once per repo.

Reply:
So, that's basically just one request per page-load. (that is if a repository is indeed selected, of course)

Reply:
Any updates on this PR?
@andrew

Reply:
The begin/end block here can be removed.
Both of these do the same:

```
def m
  ...
rescue
  false
end

def m
  begin
    ...
  rescue
    false
  end
end
```

ie:
```
def repo_subscribed?
  true if current_user.github_client.subscription("#{filters[:repo]}")
rescue
  false
end
```